lib/elixir/lib/kernel/error_handler.ex

```diff
@@ -3,28 +3,35 @@
 defmodule Kernel.ErrorHandler do
   @moduledoc false
 
-  def undefined_function(module, fun, args) do
+  @spec undefined_function(module, atom, list) :: term
+  def undefined_function(module, fun, args)
+      when is_atom(module) and is_atom(fun) and is_list(args) do
```

Comment:
I dont think we require these guards because its not public api and the caller isnt going to send invalid arguments.